Zend/zend_compile.c

```diff
@@ -7241,6 +7250,12 @@ static void zend_compile_prop_decl(zend_ast *ast, zend_ast *type_ast, uint32_t f
 		zend_error_noreturn(E_COMPILE_ERROR, "Properties cannot be declared abstract");
 	}
 
+	if ((ce->ce_flags & ZEND_ACC_READONLY_CLASS) && (flags & ZEND_ACC_STATIC)) {
+		zend_error_noreturn(E_COMPILE_ERROR, "Readonly class %s cannot declare static properties",
+			ZSTR_VAL(ce->name)
+		);
+	}
```

Comment:
Do we need this check? I believe it should already error when we set the readonly flag on the property.

Reply:
Yes, you are right. IIRC I added this check for a more explicit error message in order not to confuse people why adding a "normal" static property is an readonly error. Or Do you think the original message would be better to display in this case?

Reply:
As I'll be out this week, I went ahead and merged the PR. I'm going to commit a follow up if you insist on changing the error message.

Reply:
No strong opinion -- I mainly thought it weird that you've added an explicit check for this one case, while other readonly-related error conditions (e.g. no type) use the generic messages.

Reply:
Hmm, you must be right. I'll remove this when I am back.